src/concrete/ml/onnx/convert.py

```diff
@@ -208,7 +221,13 @@ def get_equivalent_numpy_forward_from_onnx(
             f"Available ONNX operators: {', '.join(sorted(IMPLEMENTED_ONNX_OPS))}"
         )
 
+    # Compute the LSB to be remove manually while waiting for the truncate feature
+    # FIXME: https://github.com/zama-ai/concrete-ml/issues/397
+    if use_rounding:
+        assert q_x is not None, "A quantized and representative inputset  is needed to use rounding"
+        lsbs_to_remove = compute_lsb_to_remove_for_trees(equivalent_onnx_model, q_x)
```

Comment:
I find it weird that we have a `compute_lsb_to_remove_for_trees` mentioning `trees` only in a general function like `get_equivalent_numpy_forward_from_onnx` ! 

What I mean is that, for example, we could use `get_equivalent_numpy_forward_from_onnx` with `use_rounding=True` on a torch model that has nothing to do with tree-based models

looking at the code, I believe this function is indeed specifically designed for tree-based models, so maybe instead of having a `use_rounding` then we should have a `_use_rounding_for_trees` or something like that ! I'm making it private because I have a feelink it's going to change once the truncated feature appears

but maybe I'm missing something important here !

Reply:
I agree. It feels a bit weird to have so many modifications to the core files in the /src/concrete/ml/onnx/. I don't think they should be used to implement specific features of specific models as it has nothing to do with their main purpose. Maybe it's time for a deeper discussion on how we could handle trees better in CML such that features like this this one can be implemented in a more object oriented way without touching at the onnx files

Reply:
`get_equivalent_numpy_forward_from_onnx` is indeed only used for trees. We wouldn't want to use it on anything else. Then it could maybe be called something else like `get_equivalent_numpy_forward_from_tree_onnx`